payments/views.py

```diff
@@ -15,18 +15,49 @@ class WorkshopList(ListView):
     context_object_name = 'upcoming_workshops'
 
 
-class WorkshopDetail(DetailView):
+class WorkshopDetail(FormMixin, DetailView):
     model = Workshop
+    form_class = OrderForm
     context_object_name = 'workshop'
     template_name = 'payments/detail.html'
 
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs['workshop'] = self.object
+        return kwargs
+
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
-        context['rates'] = self.object.rate_set.order_by('price')
-        context['order_form'] = OrderForm(workshop=self.object)
+        context['form'] = self.get_form()
+        rates = []
+        for rate in self.object.rate_set.order_by('price'):
+            field = context['form'][rate.name]
+            rates.append({'field': field, 'name': rate.name, 'price':
+                          rate.price})
```

Comment:
Nice 👍  This seems exactly like what we were trying earlier that wouldn't work! Guess it was just the trying to do it internally to the Form class was what was breaking it.